addons/website/models/website.py

```diff
@@ -293,6 +301,8 @@ def _unlink_except_default_website(self):
     def unlink(self):
         self._remove_attachments_on_website_unlink()
 
+        self.website_design_id.unlink()
```

Comment:
- Maybe it would be more logical to do that with a cascade delete on the website.design model (and thus a many2one website_id on the design model).
- Maybe also instead of a many2one to website.design, it could actually be a one2many. That would circle around the need to define a compute function to find the right website.design record and it would actually gives us more flexibility in the future.